src/syscall.h

```diff
@@ -3,9 +3,12 @@
 #ifndef SYSCALL_H_
 #define SYSCALL_H_
 
+#if HAVE_LINUX_AIO_ABI_H
 #include <linux/aio_abi.h>
 #include <signal.h>
 #include <time.h>
+#endif
+
 #if HAVE_LINUX_IO_URING_H
 #include <linux/io_uring.h>
 #endif
```

Comment:
No sure but don't you forget to add those declarations below under an #if?
I tried your branch on OSX and it complains about the `aoi` types.